src/code/z_lifemeter.c

```diff
@@ -1,59 +1,147 @@
+ * hearts is the red color of the hearts.
+ ***/
+
+#define HEART_PRIM_R 255
+#define HEART_PRIM_G 70
+#define HEART_PRIM_B 50
+
+#define HEART_ENV_R 50
+#define HEART_ENV_G 40
+#define HEART_ENV_B 60
+
+#define HEART_DOUBLE_DEFENSE_PRIM_R 255
+#define HEART_DOUBLE_DEFENSE_PRIM_G 255
+#define HEART_DOUBLE_DEFENSE_PRIM_B 255
+
+#define HEART_DOUBLE_DEFENSE_ENV_R 200
+#define HEART_DOUBLE_DEFENSE_ENV_G 0
+#define HEART_DOUBLE_DEFENSE_ENV_B 0
+
+/***
+ * The burn and drown colors listed here are unused. Prerelease footage of the game confirms that at one
+ * point in development the orange color was to be used while taking damage from hot environments.
+ * Based on this, we can assume that the blue heart color was to be used while drowning.
+ * In the final game these environments only have a timer and do not damage you continuously.
+ ***/
+
+#define HEART_BURN_PRIM_R 255
+#define HEART_BURN_PRIM_G 190
+#define HEART_BURN_PRIM_B 0
+
+#define HEART_PRIM_BURN_ENV_R 255
+#define HEART_PRIM_BURN_ENV_G 0
+#define HEART_PRIM_BURN_ENV_B 0
+
+#define HEART_DROWN_PRIM_R 100
+#define HEART_DROWN_PRIM_G 100
+#define HEART_DROWN_PRIM_B 255
+
+#define HEART_DROWN_ENV_R 0
+#define HEART_DROWN_ENV_G 0
+#define HEART_DROWN_ENV_B 255
+
+s16 sHeartPrimColors[3][3] = {
+    { HEART_PRIM_R, HEART_PRIM_G, HEART_PRIM_B },
+    { HEART_BURN_PRIM_R, HEART_BURN_PRIM_G, HEART_BURN_PRIM_B },    // unused
+    { HEART_DROWN_PRIM_R, HEART_DROWN_PRIM_G, HEART_DROWN_PRIM_B }, // unused
+};
+
+s16 sHeartEnvColors[3][3] = {
+    { HEART_ENV_R, HEART_ENV_G, HEART_ENV_B },
+    { HEART_PRIM_BURN_ENV_R, HEART_PRIM_BURN_ENV_G },            // unused
+    { HEART_DROWN_ENV_R, HEART_DROWN_ENV_G, HEART_DROWN_ENV_B }, // unused
+};
+
+s16 sHeartPrimFactors[3][3] = {
+    { 0, 0, 0 },
+    { 0, 120, -50 },   // unused
+    { -155, 30, 205 }, // unused
+};
+
+s16 sHeartEnvFactors[3][3] = {
+    { 0, 0, 0 },
+    { 205, -40, -60 }, // unused
+    { -50, -40, 195 }, // unused
+};
+
+s16 sHeartDDPrimColors[3][3] = {
+    { HEART_DOUBLE_DEFENSE_PRIM_R, HEART_DOUBLE_DEFENSE_PRIM_G, HEART_DOUBLE_DEFENSE_PRIM_B },
+    { HEART_BURN_PRIM_R, HEART_BURN_PRIM_G, HEART_BURN_PRIM_B },    // unused
+    { HEART_DROWN_PRIM_R, HEART_DROWN_PRIM_G, HEART_DROWN_PRIM_B }, // unused
+};
+
+s16 sHeartDDEnvColors[3][3] = {
+    { HEART_DOUBLE_DEFENSE_ENV_R, HEART_DOUBLE_DEFENSE_ENV_G, HEART_DOUBLE_DEFENSE_ENV_B },
+    { HEART_PRIM_BURN_ENV_R, HEART_PRIM_BURN_ENV_G, HEART_PRIM_BURN_ENV_B }, // unused
+    { HEART_DROWN_ENV_R, HEART_DROWN_ENV_G, HEART_DROWN_ENV_B },             // unused
+};
+
+s16 sHeartDDPrimFactors[3][3] = {
+    { 0, 0, 0 },
+    { 0, -65, -255 },  // unused
+    { -155, -155, 0 }, // unused
+};
+
+s16 sHeartDDEnvFactors[3][3] = {
+    { 0, 0, 0 },
+    { 55, 0, 0 },     // unused
+    { -200, 0, 255 }, // unused
+};
+
+s16 sHeartPrimCurrent1[3];
+s16 sHeartEnvCurrent1[3];
+s16 sHeartPrimCurrent2[2][3];
+s16 sHeartEnvCurrent2[2][3];
```

Comment:
I believe these 4 arrays are specifically for DD hearts, as opposed to the colors in interfaceCtx which are normal hearts. Also I think Current2 is what you called beatingHeart in interfaceCtx, so it would be a good idea to name them consistently.